solvers/cuml.py

```diff
@@ -1,21 +1,18 @@
 from benchopt import BaseSolver, safe_import_context
-from benchopt.utils.sys_info import get_cuda_version
-
-cuda_version = get_cuda_version()
+from benchopt.helpers.requires_gpu import requires_gpu
+from benchopt.stopping_criterion import SufficientProgressCriterion
 
 with safe_import_context() as import_ctx:
-    if cuda_version is not None:
-        cuda_version = cuda_version.split("cuda_", 1)[1][:4]
-    else:
-        raise ImportError("cuml solver needs a nvidia GPU.")
-
     import numpy as np
     from scipy import sparse
+    cuda_version = requires_gpu()
 
-    import cudf
-    import cupy as cp
-    import cupyx.scipy.sparse as cusparse
-    from cuml.linear_model import Lasso
+    if cuda_version is not None:
+
+        import cudf
+        import cupy as cp
+        import cupyx.scipy.sparse as cusparse
+        from cuml.linear_model import Lasso
```

Comment:
The issue here is that you need `cuda_version` in requirements. not sure what is the best way to handle this case.

Reply:
Yes, my main issue is when installing on a cluster, but I guess we should provide a way to specify this in install.
This is for later, your fix is great for this one.